src/DotNetWorker/Broker/FunctionBroker.cs

```diff
@@ -70,19 +70,15 @@ public async Task<InvocationResponse> InvokeAsync(FunctionInvocation invocation)
                     var parameterBindings = executionContext.OutputBindings;
                     var result = executionContext.InvocationResult;
 
-                    foreach (var paramBinding in parameterBindings)
+                    // TODO: ParameterBinding shouldn't happen here
+                    foreach (var binding in parameterBindings)
                     {
-                        // TODO: ParameterBinding shouldn't happen here
-
-                        foreach (var binding in executionContext.OutputBindings)
+                        var parameterBinding = new ParameterBinding
                         {
-                            var parameterBinding = new ParameterBinding
-                            {
-                                Name = binding.Key,
-                                Data = binding.Value.ToRpc(_workerOptions.Value.Serializer)
-                            };
-                            response.OutputData.Add(parameterBinding);
-                        }
+                            Name = binding.Key,
+                            Data = await binding.Value.ToRpcAsync(_workerOptions.Value.Serializer)
```

Comment:
Made this to `ToRpcAsync` as that handles `HttpResponseData` as well

Reply:
nit: delete `ToRpc` method to avoid future accidental bugs?

Reply:
`ToRpc` is still used by `ToRpcHttpAsync` on the Body and such. I think we may want to clean or rename some of these functions, as I agree `ToRpc` and `ToRpcAsync` shouldn't have a ton of difference (other than async).

I will leave that for later though, as it may be invasive and can't simply remove the `ToRpc` without consequences.

Reply:
Yes, that should be done in a separate PR as I went down the same path when introduced `ToRpcAsync`. That should be gone.

Reply:
Yes, as mentioned above, different PR and we'll clean that up.